pkg/storage/sqlcommon/sqlcommon.go

```diff
@@ -445,13 +447,27 @@ func ReadAuthorizationModel(ctx context.Context, dbInfo *DBInfo, store, modelID
 }
 
 // IsReady returns true if the connection to the datastore is successful
-func IsReady(ctx context.Context, db *sql.DB) (bool, error) {
+func IsReady(ctx context.Context, db *sql.DB) (storage.ReadinessStatus, error) {
 	ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
 	defer cancel()
 
 	if err := db.PingContext(ctx); err != nil {
-		return false, err
+		return storage.ReadinessStatus{}, err
 	}
 
-	return true, nil
+	revision, err := goose.GetDBVersion(db)
+	if err != nil {
+		return storage.ReadinessStatus{}, err
+	}
+
+	if revision < build.MinimumSupportedDatastoreSchemaRevision {
+		return storage.ReadinessStatus{
+			Message: fmt.Sprintf("datastore requires migrations: at revision '%d', but requires '%d'. Run 'openfga migrate'.", revision, build.MinimumSupportedDatastoreSchemaRevision),
+			IsReady: false,
+		}, nil
+	}
```

Comment:
This code isn't covered. Can you add a test for this? You can copy the one I wrote here https://github.com/openfga/openfga/blob/8e36f05d7f55d3561cd298e807829b844c2b09df/cmd/migrate/migrate_test.go#L18-L48

Reply:
Pushed up.